plugin-parser.js

```diff
@@ -0,0 +1,41 @@
+var path = require("path");
+var fs  = require("fs");
+
+exports.loadPlugins = function(callback) {
+    var plugins = [];
+    var pluginFolder = path.resolve(__dirname + path.sep + 'plugins');
+    
+    fs.readdir(pluginFolder, function(err, files) {
+        if (err) {
+            console.log(err);
+            return;
+        }
+
+        // consider revamp to async forEach.
+        // this method causes considerable startup delay
+        var callbackCounter = files.length;
```

Comment:
I agree that we can improve this, but I think it can be done after the code here is in tree.